src/components/views/avatars/MemberAvatar.tsx

```diff
@@ -46,100 +47,85 @@ interface IProps extends Omit<React.ComponentProps<typeof BaseAvatar>, "name" |
     hideTitle?: boolean;
 }
 
-interface IState {
-    name: string;
-    title: string;
-    imageUrl?: string;
-}
-
-export default class MemberAvatar extends React.PureComponent<IProps, IState> {
-    public static defaultProps = {
-        width: 40,
-        height: 40,
-        resizeMethod: 'crop',
-        viewUserOnClick: false,
-    };
-
-    constructor(props: IProps) {
-        super(props);
-
-        this.state = MemberAvatar.getState(props);
-    }
+export default function MemberAvatar({
+    width,
+    height,
+    resizeMethod = 'crop',
+    viewUserOnClick,
+    ...props
+}: IProps) {
+    const cli = useContext(MatrixClientContext);
+    const card = useContext(CardContext);
+    const roomContext = useContext(RoomContext);
+
+    const [name, setName] = useState<string | undefined>();
+    const [title, setTitle] = useState<string | undefined>();
+    const [imageUrl, setImageUrl] = useState<string | undefined>();
+    const [userId, setUserId] = useState<string | undefined>();
+
+    useEffect(() => {
```

Comment:
I suggest to create a custom hook here to improve the readability of the component